### PR TITLE
Feature/add ldap gem and config

### PR DESCRIPTION
**Pull request by syl-p**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class ApplicationController < ActionController::API
+  rescue_from DeviseLdapAuthenticatable::LdapException do |exception|
+    render :text => exception, :status => 500
```
Comment (brunto) [inline app/controllers/application_controller.rb:3]:
```suggestion
    render text: exception, status: 500
```

File: app/models/user.rb [hunk 2]
```diff
@@ -22,6 +22,36 @@ class User < ApplicationRecord
            foreign_key: :resource_owner_id,
            dependent: :destroy
 
+  def get_ldap_email
```
Comment (brunto) [inline app/models/user.rb:25]: A method should not start with the word `get`

File: config/initializers/doorkeeper.rb [hunk 3]
```diff
@@ -347,11 +347,34 @@
...
+    user = nil
+
+    # LDAP IS ACTIVE
+    if ENV['DEVISE_LDAP_LOGGER'].present?
+      # Check LDAP credentials
+      if User.check_ldap_credentials(login, password) # ldap valide
+        user = User.find_for_authentication(login: login)
+        #  Check if user exists in database
+        if user.present?
+        else
```
Comment (brunto) [inline config/initializers/doorkeeper.rb:361]: If the `if` is empty, you should change to:
```suggestion
        if user.blank?
```

File: config/initializers/doorkeeper.rb [hunk 4]
```diff
@@ -347,11 +347,34 @@
...
+          # Create user in database
+          user = User.create_with_ldap(login)
+        end
+      end
+    end
+
+    # Normal auth
+    if user.nil?
```
Comment (brunto) [inline config/initializers/doorkeeper.rb:369]:
```suggestion
    if user.blank?
```

File: db/migrate/20211118113021_add_login_to_user_table.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class AddLoginToUserTable < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :login, :string
```
Comment (brunto) [inline db/migrate/20211118113021_add_login_to_user_table.rb:3]: You must add an index to keep `login` unique.
We also need to add, inside the User model, a validation to verify the uniqueness of the login field.

**Review by brunto — CHANGES_REQUESTED** [review]

File: db/migrate/20211118113021_add_login_to_user_table.rb (hunk 5 above)
Comment (brunto) [reply db/migrate/20211118113021_add_login_to_user_table.rb:3]: @syl-p

Comment (syl-p) [reply db/migrate/20211118113021_add_login_to_user_table.rb:3]: Pas eu encore le temps de travailler dessus.

**Review by brunto — CHANGES_REQUESTED** [review]
Update the schema with rake

**Pull request merged**